### PR TITLE
refactor(fields): upgrade component - INNO-609

### PR DESCRIPTION
**Pull request by emeryro**
Upgrade of field components.
There were some modifiers to alter text display, so I added utilities to handle it (more flexible).
I also had to update headings, to fix margin display issues.

### CODE REVIEW

File: framework/components/ecl-fields/package.json [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
-  "description": "",
-  "main": "_fields.scss",
-  "style": "_fields.scss",
   "author": "European Commission",
   "license": "EUPL-1.1",
-  "peerDependencies": {
-    "@ec-europa/ecl-social-icons": "^0.1.0"
-  }
+  "version": "0.3.0",
+  "description": "ECL Expandables",
```
Comment (kalinchernev) [inline framework/components/ecl-fields/package.json]: Wrong description.

File: framework/components/ecl-fields/_fields.scss [hunk 2]
```diff
@@ -1,80 +0,0 @@
...
-
-.field {
-  margin-bottom: 0.556em;
-
-  + .field--aligned,
-  &:last-child {
-    margin-bottom: 0;
-  }
-
-  .field__item--expired {
```
Comment (kalinchernev) [inline framework/components/ecl-fields/_fields.scss]: Missing in migrated version.

File: framework/components/ecl-fields/_fields.scss [hunk 3]
```diff
@@ -1,80 +0,0 @@
...
-
-  .field__label {
-    font-size: map-get($ecl-font-size, 'xs');
-    font-weight: 600;
-
-    h4 {
-      padding: 0;
-    }
-
-    a:focus {
```
Comment (kalinchernev) [inline framework/components/ecl-fields/_fields.scss]: Missing in migrated version.

File: framework/components/ecl-fields/ecl-fields.config.js [hunk 4]
```diff
@@ -1,17 +1,33 @@
...
+    `;
+  },
   tags: ['atom'],
   variants: [
     {
       name: 'default',
       label: 'Default',
+      context: {
+        label: 'Name',
+        body: 'John Doe',
```
Comment (kalinchernev) [inline framework/components/ecl-fields/ecl-fields.config.js]: I think they will ask about the examples with icons, links and non-textual content they demo in current implementation.

File: framework/index.scss [hunk 5]
```diff
@@ -117,3 +117,4 @@
 @import 'utilities/ecl-padding';
 @import 'utilities/ecl-ratio';
 @import 'utilities/ecl-screen-reader';
+@import 'utilities/ecl-text';
```
Comment (kalinchernev) [inline framework/index.scss]: Was it included to make an example with the `--expired` modifier? I don't see any other reason that would be included in this ticket.

File: framework/utilities/ecl-text/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@ec-europa/ecl-text",
+  "author": "European Commission",
+  "license": "EUPL-1.1",
+  "version": "0.1.0",
+  "description": "ECL Text",
```
Comment (kalinchernev) [inline framework/utilities/ecl-text/package.json]: Missing tests for this one.

**Comment by kalinchernev** [comment]
Few errors from the CI:

```
1) fields field with plain state should match the reference screenshot:
expected screenshots to match reference
running chrome (v56.0) on Windows 10
Check out job at https://saucelabs.com/tests/022340ec150c4855b97ed54ef4723510
running safari (v9.0) on OS X 10.11
Check out job at https://saucelabs.com/tests/76355b8c9ca44e4dacfda275a778598f
running firefox (v46.0) on Windows 7
Check out job at https://saucelabs.com/tests/fc2f1486aa6e4d58bb24e033d25140b0
running internet explorer (v11.0) on Windows 7
Check out job at https://saucelabs.com/tests/b7c8914831c94a83b155de837b49cf9a
AssertionError: expected screenshots to match reference
    at Context.it (/home/travis/build/ec-europa/europa-component-library/test/functional/fields.js:28:32)
    at Promise.F (/home/travis/build/ec-europa/europa-component-library/node_modules/core-js/library/modules/_export.js:35:28)
2) ecl-page-headers--corporate should match the reference screenshot:
expected screenshots to match reference
running chrome (v56.0) on Windows 10
Check out job at https://saucelabs.com/tests/1e24d7a21feb4229879d067d0f0db675
running internet explorer (v11.0) on Windows 7
Check out job at https://saucelabs.com/tests/dabb71722ecd49098a44de2d5a244290
running firefox (v46.0) on Windows 7
Check out job at https://saucelabs.com/tests/aa8b3130f0eb4820a8aeb68c51dcb63c
running safari (v9.0) on OS X 10.11
Check out job at https://saucelabs.com/tests/8d2ea08665a740368a1cddc4b7e34e2c
AssertionError: expected screenshots to match reference
    at Context.it (/home/travis/build/ec-europa/europa-component-library/test/functional/page-headers.js:31:30)
    at Promise.F (/home/travis/build/ec-europa/europa-component-library/node_modules/core-js/library/modules/_export.js:35:28)
3) ecl-page-headers--improved-complete should match the reference screenshot:
expected screenshots to match reference
running chrome (v56.0) on Windows 10
Check out job at https://saucelabs.com/tests/1e24d7a21feb4229879d067d0f0db675
running internet explorer (v11.0) on Windows 7
Check out job at https://saucelabs.com/tests/dabb71722ecd49098a44de2d5a244290
running firefox (v46.0) on Windows 7
Check out job at https://saucelabs.com/tests/aa8b3130f0eb4820a8aeb68c51dcb63c
running safari (v9.0) on OS X 10.11
Check out job at https://saucelabs.com/tests/8d2ea08665a740368a1cddc4b7e34e2c
AssertionError: expected screenshots to match reference
    at Context.it (/home/travis/build/ec-europa/europa-component-library/test/functional/page-headers.js:31:30)
    at Promise.F (/home/travis/build/ec-europa/europa-component-library/node_modules/core-js/library/modules/_export.js:35:28)
```

File: framework/components/ecl-fields/_fields.scss (hunk 2 above)
Comment (emeryro) [inline framework/components/ecl-fields/_fields.scss]: It is managed by utilities

File: framework/components/ecl-fields/_fields.scss (hunk 3 above)
Comment (emeryro) [inline framework/components/ecl-fields/_fields.scss]: Every link should use the ecl-link ckass, which managed focus and hover states

File: framework/components/ecl-fields/ecl-fields.config.js (hunk 4 above)
Comment (emeryro) [inline framework/components/ecl-fields/ecl-fields.config.js]: Well here we have an issue.
With our current implementation, we don't have multiple examples of each components. Maybe we could create some example pages, but the component itself should remain as simple as possible (for composition)

File: framework/index.scss (hunk 5 above)
Comment (emeryro) [inline framework/index.scss]: Yes, that's the reason

File: framework/utilities/ecl-text/package.json (hunk 6 above)
Comment (emeryro) [inline framework/utilities/ecl-text/package.json]: We don't have test for other utilities, but indeed we could add some.

**Comment by kalinchernev** [comment]
@emeryro thanks for the updates, the template is much better. I restarted the CI and will merge when it's green.

**Pull request merged**